src/messaging/messaging-types.ts

```diff
@@ -699,10 +747,73 @@ function validateAndroidNotification(notification: AndroidNotification) {
     titleLocArgs: 'title_loc_args',
     channelId: 'channel_id',
     imageUrl: 'image',
+    eventTimestamp: 'event_time',
+    localOnly: 'local_only',
+    priority: 'notification_priority',
+    vibrateTimingsMillis: 'vibrate_timings',
+    defaultVibrateTimings: 'default_vibrate_timings',
+    defaultSound: 'default_sound',
+    lightSettings: 'light_settings',
+    defaultLightSettings: 'default_light_settings',
+    notificationCount: 'notification_count',
   };
   renameProperties(notification, propertyMappings);
 }
 
+/**
+ * Checks if the given LightSettings object is valid. The object must have valid color and
+ * light on/off duration parameters. If successful, transforms the input object by renaming
+ * keys to valid Android keys.
+ *
+ * @param {LightSettings} lightSettings An object to be validated.
+ */
+function validateLightSettings(lightSettings: LightSettings) {
+  if (typeof lightSettings === 'undefined') {
+    return;
+  } else if (!validator.isNonNullObject(lightSettings)) {
+    throw new FirebaseMessagingError(
+      MessagingClientErrorCode.INVALID_PAYLOAD, 'android.notification.lightSettings must be a non-null object');
+  }
+
+  if (!validator.isNumber(lightSettings.lightOnDurationMillis) || lightSettings.lightOnDurationMillis < 0) {
+    throw new FirebaseMessagingError(
+      MessagingClientErrorCode.INVALID_PAYLOAD,
+      'android.notification.lightSettings.lightOnDurationMillis must be a non-negative duration in milliseconds');
+  }
+  const durationOn = transformMillisecondsToSecondsString(lightSettings.lightOnDurationMillis);
+  (lightSettings as any).lightOnDurationMillis = durationOn;
+
+  if (!validator.isNumber(lightSettings.lightOffDurationMillis) || lightSettings.lightOffDurationMillis < 0) {
+    throw new FirebaseMessagingError(
+      MessagingClientErrorCode.INVALID_PAYLOAD,
+      'android.notification.lightSettings.lightOffDurationMillis must be a non-negative duration in milliseconds');
+  }
+  const durationOff = transformMillisecondsToSecondsString(lightSettings.lightOffDurationMillis);
+  (lightSettings as any).lightOffDurationMillis = durationOff;
+
+  if (!validator.isString(lightSettings.color) ||
+    (!/^#[0-9a-fA-F]{6}$/.test(lightSettings.color) && !/^#[0-9a-fA-F]{8}$/.test(lightSettings.color))) {
+    throw new FirebaseMessagingError(
+      MessagingClientErrorCode.INVALID_PAYLOAD,
+      'android.notification.lightSettings.color must be in the form #RRGGBB or #RRGGBBAA format');
+  }
+  lightSettings.color += lightSettings.color.length === 7 ? 'FF' : '';
```

Comment:
@hiranya911:
Do you think it would be better to generate the color string and assigned to a temp variable without updating the `LightSettings` object directly here?
```
const colorString = lightSettings.color.length === 7 ? lightSettings.color + 'FF' : lightSettings.color;
// populate color object
(lightSettings as any).color = colorString;
```

Reply:
Yeah, that looks cleaner.